Projects/Applications/FreeRTOS/FreeRTOS_LoRaWAN/Core/App/Src/receive.c

```diff
@@ -35,16 +32,42 @@ void Interpret_Message(uint8_t *Buffer, uint8_t BufferSize) {
 			APP_LOG(TS_OFF, VLEVEL_M, "Select motion sensor!\n");
 			break;
 		case CHANGE_BRIGHTNESS:
-			if (BufferSize <= MESSAGE_MIN_BYTES) {
+			if (buffer_size < MESSAGE_MIN_BYTES + BRIGHTNESS_PARAMS_BYTE_COUNT) {
 				APP_LOG(TS_OFF, VLEVEL_M,
 						"Brightness command does not include brightness\n");
 			} else {
 				APP_LOG(TS_OFF, VLEVEL_M, "Change brightness to %02X!\n",
-						Buffer[PARAMETERS_START_BYTE]);
+						buffer[PARAMETERS_START_BYTE]);
 			}
 			break;
 		case SEND_BATTERY_STATUS:
-			APP_LOG(TS_OFF, VLEVEL_M, "Report battery state!\n");
+			APP_LOG(TS_OFF, VLEVEL_M, "Report battery state!\r\n");
+			{
+				const uint8_t params[] = { SEND_BATTERY_STATUS, Get_Battery_Level() };
+				Tx_Set_Buffer(RESPONSE_OUT_WITH_DATA, RESPONDING_TO_INSTRUCTION, &params, 2);
+			}
+			break;
+		case SET_BATTERY_VREF:
+			APP_LOG(TS_OFF, VLEVEL_M, "Setting battery min and max vref\r\n");
```

Comment:
voor debug redenen is het misschien handig om hier ook de meegegeven waardes (in decimaal) te printen